### ISSUE TITLE
How do you run mlpm behind a proxy?

### ISSUE THREAD

**Issue by peetkes** [joemfb/mlpm#16]
I wanted to run mlpm behind a proxy so I adjusted some npm settigs to be able to install mlpm on my windows machine.
That wiorked out well but now mlpm will not work.
What kind of proxy settings do I need on windows?

**Comment by grtjn** [comment]
Maybe @joemfb knows better, but the code seems to be using the npm package `request` to make backend calls. The docs for that seem to indicate it should respect env vars: https://github.com/request/request#controlling-proxy-behaviour-using-environment-variables

Worth a try?

**Comment by joemfb** [comment]
Those ENV vars seem like a good bet. If that doesn't work, you could try creating local DNS bindings for http://registry.demo.marklogic.com in /etc/hosts or the windows equivalent. 

Note: the registry URL isn't configureable (although it should be) -- it's just hardcoded here: https://github.com/joemfb/mlpm/blob/master/lib/api.js#L5.

**Comment by peetkes** [comment]
the env vars do not do it for me. 
It will give me ECONNREFUSED so I'm passed the timeout I got before the env settings

How does the local DNS binding work? How do I configure that?

**Comment by joemfb** [comment]
The details would depend on what kind of proxy you're using, and with what settings. I'm assuming that you are unable to `ping registry.demo.marklogic.com` from behind the proxy, but are able to from "in front" of the proxy.

You might be able to configure your proxy to map some local name, or just to forward a specific port, to registry.demo.marklogic.com. If it needs to be a name, you could bind that name on your machine by using the hosts file: https://en.wikipedia.org/wiki/Hosts_(file).

In other words, mlpm will connect to registry.demo.marklogic.com on port 80. It doesn't not have any specific facility for redirection, proxying, tunneling, etc.

That's about all I have off the top of my head. If you'd like to post a sanitized copy of the npm config you're using, I'll try to look further.
